app/src/main/kotlin/com/simplemobiletools/voicerecorder/extensions/Activity.kt

```diff
@@ -0,0 +1,173 @@
+package com.simplemobiletools.voicerecorder.extensions
+
+import android.content.ContentValues
+import android.provider.MediaStore
+import android.provider.MediaStore.Audio.Media
+import androidx.core.net.toUri
+import com.simplemobiletools.commons.activities.BaseSimpleActivity
+import com.simplemobiletools.commons.extensions.deleteFile
+import com.simplemobiletools.commons.extensions.getParentPath
+import com.simplemobiletools.commons.extensions.toFileDirItem
+import com.simplemobiletools.commons.helpers.*
+import com.simplemobiletools.commons.models.FileDirItem
+import com.simplemobiletools.voicerecorder.models.Recording
+import java.io.File
+
+fun BaseSimpleActivity.deleteRecordings(recordingsToRemove: Collection<Recording>, callback: (Boolean) -> Unit) {
+    when {
+        isRPlus() -> {
+            val fileUris = recordingsToRemove.map { recording ->
+                "${Media.EXTERNAL_CONTENT_URI}/${recording.id.toLong()}".toUri()
+            }
+
+            deleteSDK30Uris(fileUris, callback)
+        }
+
+        isQPlus() -> {
+            recordingsToRemove.forEach {
+                val uri = Media.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
+                val selection = "${Media._ID} = ?"
+                val selectionArgs = arrayOf(it.id.toString())
+                val result = contentResolver.delete(uri, selection, selectionArgs)
+
+                if (result == 0) {
+                    deleteFile(File(it.path).toFileDirItem(this))
```

Comment:
theres too much going on, lets move the filedir to a separate file like `val fileDirItem = ...` please